src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicNavigationBarItemService.vb

```diff
@@ -488,6 +482,19 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.NavigationBar
             Return New VirtualTreePoint(location.SourceTree, location.SourceTree.GetText(cancellationToken), location.SourceSpan.Start)
         End Function
 
+        Private Shared Function GetSourceNavigationLocation(document As Document, symbol As ISymbol, cancellationToken As CancellationToken) As Location
+            Dim sourceLocations = symbol.Locations.Where(Function(l) l.IsInSource)
```

Comment:
The only actual change is to only consider `IsInSource` locations. Extracted this bit to a method to make sure the setting of the "location" and "sourceLocations" locals didn't leak out to the rest of the method.